daphne/src/vdaf/mod.rs

```diff
@@ -172,17 +173,32 @@ impl VdafConfig {
         version: DapVersion,
     ) -> Result<Report, DapError> {
         let mut rng = thread_rng();
+        let report_extensions = match version {
+            DapVersion::Draft02 => extensions.clone(),
+            _ => vec![],
+        };
         let metadata = ReportMetadata {
             id: ReportId(rng.gen()),
             time,
-            extensions,
+            extensions: report_extensions,
         };
 
         let public_share = Vec::new();
-        let encoded_input_shares = match self {
+        let mut encoded_input_shares = match self {
             Self::Prio3(prio3_config) => prio3_shard(prio3_config, measurement)?,
             Self::Prio2 { dimension } => prio2_shard(*dimension, measurement)?,
         };
+        if version != DapVersion::Draft02 {
+            let mut encoded: Vec<Vec<u8>> = Vec::new();
+            for share in encoded_input_shares.drain(..) {
```

Comment:
I think into_iter() is more conventional here.
```suggestion
            for share in encoded_input_shares.into_iter() {
```